Create generic solution for casting bool to int*

main.c
```
int main() {
    char i1 = (1 == 2);
    short i2 = (1 == 2);
    int i3 = (1 == 2);
    long i4 = (1 == 2);
    long long i5 = (1 == 2);
    return (i1==0) && (i2==0) && (i3==0) && (i4==0) && (i5==0);
}
```

Current go output:
```
package main

import "os"
import "github.com/elliotchance/c2go/noarch"

type __int128_t int64
type __uint128_t uint64
type __builtin_ms_va_list []byte

func main() {
        __init()
        var i1 byte = noarch.BoolToByte((1 == 2))
        var i2 int16 = noarch.BoolToInt16((1 == 2))
        var i3 int = noarch.BoolToInt((1 == 2))
        var i4 int32 = noarch.BoolToInt32((1 == 2))
        var i5 int64 = noarch.BoolToInt64((1 == 2))
        os.Exit(noarch.BoolToInt((i1 == byte(0)) && (i2 == int16(0)) && (i3 == 0) && (i4 == int32(0)) && (i5 == int64(0))))
}
func __init() {
}
```

go build main.go
```
# command-line-arguments
./main.go:14:16: undefined: noarch.BoolToByte
./main.go:15:17: undefined: noarch.BoolToInt16
./main.go:17:17: undefined: noarch.BoolToInt32
./main.go:18:17: undefined: noarch.BoolToInt64
```

proposed go code
```
package main

import "os"

type __int128_t int64
type __uint128_t uint64
type __builtin_ms_va_list []byte

func main() {
        __init()
        var i1 byte = byte(map[bool]int{false: 0, true: 1}[(1 == 2)])
        var i2 int16 = int16(map[bool]int{false: 0, true: 1}[(1 == 2)])
        var i3 int = map[bool]int{false: 0, true: 1}[(1 == 2)]
        var i4 int32 = int32(map[bool]int{false: 0, true: 1}[(1 == 2)])
        var i5 int64 = int64(map[bool]int{false: 0, true: 1}[(1 == 2)])
        os.Exit(map[bool]int{false: 0, true: 1}[(i1 == byte(0)) && (i2 == int16(0)) && (i3 == 0) && (i4 == int32(0)) && (i5 == int64(0))])
}
func __init() {
}
```

--- a/tests/cast.c
+++ b/tests/cast.c
@@ -14,11 +14,23 @@ void test_cast()
     is_eq(d, 1.0);
 }
 
+void test_castbool()
+{
+    char i1 = (1 == 1);
+    short i2 = (1 == 1);
+    int i3 = (1 == 1);
+    long i4 = (1 == 0);
+    long long i5 = (1 == 0);
+
+    is_eq((i1==1) && (i2==1) && (i3==1) && (i4==0) && (i5==0), 1);
+}
+
 int main()
 {
-    plan(2);
+    plan(3);
 
     START_TEST(cast)
+    START_TEST(castbool)
 
     done_testing();
 }

--- a/types/cast.go
+++ b/types/cast.go
@@ -183,7 +183,8 @@ func CastExpr(p *program.Program, expr goast.Expr, cFromType, cToType string) (g
 			return e, nil
 		}
 		if fromType == "bool" && toType == v {
-			e := util.NewGoExpr(`map[bool]int{false: 0, true: 1}[false]`)
+			e := util.NewGoExpr(`map[bool]int{false: 0, true: 1}[replaceme]`)
+			// Swap replaceme with the current expression
 			e.(*goast.IndexExpr).Index = expr
 			return CastExpr(p, e, "int", cToType)
 		}

--- a/types/cast.go
+++ b/types/cast.go
@@ -60,7 +60,10 @@ func GetArrayTypeAndSize(s string) (string, int) {
 //    a bug. It is most useful to do this when dealing with compound types like
 //    FILE where those function probably exist (or should exist) in the noarch
 //    package.
-func CastExpr(p *program.Program, expr goast.Expr, fromType, toType string) (goast.Expr, error) {
+func CastExpr(p *program.Program, expr goast.Expr, cFromType, cToType string) (goast.Expr, error) {
+	fromType := cFromType
+	toType := cToType
+
 	// Replace for specific case of fromType for darwin:
 	// Fo : union (anonymous union at sqlite3.c:619241696:3)
 	if strings.Contains(fromType, "anonymous union") {
@@ -178,6 +181,11 @@ func CastExpr(p *program.Program, expr goast.Expr, fromType, toType string) (goa
 			)
 
 			return e, nil
+		}
+		if fromType == "bool" && toType == v {
+			e := util.NewGoExpr(`map[bool]int{false: 0, true: 1}[false]`)
+			e.(*goast.IndexExpr).Index = expr
+			return CastExpr(p, e, "int", cToType)
 		}
 	}
 

--- a/types/cast_test.go
+++ b/types/cast_test.go
@@ -42,10 +42,6 @@ func TestCast(t *testing.T) {
 		// Casting to bool
 		{args{util.NewIntLit(1), "int", "bool"}, util.NewBinaryExpr(util.NewIntLit(1), token.NEQ, util.NewIntLit(0), "bool", false)},
 
-		// Casting from bool. This is a special case because C int and bool
-		// values are very commonly used interchangeably.
-		{args{util.NewIntLit(1), "bool", "int"}, util.NewCallExpr("noarch.BoolToInt", util.NewIntLit(1))},
-
 		// String types
 		// {args{"foo", "[3]char", "const char*"}, "1 != 0"},
 

--- a/util/goast.go
+++ b/util/goast.go
@@ -6,6 +6,7 @@ package util
 import (
 	"fmt"
 	goast "go/ast"
+	"go/parser"
 	"go/token"
 	"strconv"
 	"strings"
@@ -347,4 +348,16 @@ func NewFuncType(fieldList *goast.FieldList, returnType string) *goast.FuncType 
 			List: returnTypes,
 		},
 	}
+}
+
+// NewGoExpr is used to facilitate the creation of go AST
+//
+// It should not be used to transpile user code.
+func NewGoExpr(expr string) goast.Expr {
+	e, err := parser.ParseExpr(expr)
+	if err != nil {
+		panic("programming error: " + expr)
+	}
+
+	return e
 }

